client/src/pages/SignupForm.js

```diff
@@ -1,104 +1,207 @@
-import React, { useState } from "react";
-import { Grid, Button } from "@material-ui/core";
+import React, { useState, useEffect } from "react";
+import {
+  Grid,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogContentText,
+  DialogActions,
+  CircularProgress,
+  Typography,
+} from "@material-ui/core";
 import { validateEmail, validatePassword } from "../Validation";
-import { LoginComponent, PasswordComponent, NameComponent } from "./LoginSignupComponents";
-import { makeStyles } from '@material-ui/core/styles';
+import {
+  LoginComponent,
+  PasswordComponent,
+  NameComponent,
+} from "./LoginSignupComponents";
+import { makeStyles } from "@material-ui/core/styles";
+import { useHistory, Link } from "react-router-dom";
 
 const themes = makeStyles({
-    primaryButton: {
-        fontSize: "1em",
-        backgroundColor: "#FF743D",
-        color: "#fff",
-        textTransform: "none",
-        padding: "1em 3em 1em 3em",
-        margin: "1em 0 1em 0",
-        borderRadius: "0 0 0 0",
-        '&:hover': {
-            backgroundColor: '#AD6800'
-        },
-    },
-    formTitle: {
-        fontSize: "2em",
-        margin: "1em 0 1em 0",
+  primaryButton: {
+    fontSize: "1em",
+    backgroundColor: "#FF743D",
+    color: "#fff",
+    textTransform: "none",
+    padding: "1em 3em 1em 3em",
+    margin: "1em 0 1em 0",
+    borderRadius: "0 0 0 0",
+    "&:hover": {
+      backgroundColor: "#AD6800",
     },
+  },
+  formTitle: {
+    fontSize: "2em",
+    margin: "1em 0 1em 0",
+  },
+  blackFont: {
+    color: "#000",
+  },
 });
 
-
 function SignupForm(props) {
-    const [email, setEmail] = useState("");
-    const [validEmail, setValidEmail] = useState(true);
-    const [password, setPassword] = useState("");
-    const [validPassword, setValidPassword] = useState(true);
-    const [name, setName] = useState("")
-    const [validName, setValidName] = useState(true);
+  const [email, setEmail] = useState("");
+  const [validEmail, setValidEmail] = useState(true);
+  const [password, setPassword] = useState("");
+  const [validPassword, setValidPassword] = useState(true);
+  const [name, setName] = useState("");
+  const [validName, setValidName] = useState(true);
+  const [openSuccessDialog, setOpenSuccessDialog] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [signupSuccess, setSignupSuccess] = useState(false);
 
+  const history = useHistory();
 
-    const handleEmailInput = event => {
-        setEmail(event.target.value);
+  useEffect(() => {
+    if (signupSuccess) {
+      const timer = setTimeout(() => {
+        history.push("/login");
+      }, 5000);
+      return () => clearTimeout(timer);
     }
+  }, [signupSuccess]);
```

Comment:
this is good for now. In the future what we could have is an auth context, and then on log-in we just set the auth context and that would just trigger a re-render. On the sign-up and login pages we should have a check to see if a user is logged in, and if they are redirect to the search page. This would remove the need for a specific history.push since we could just do `{loggedIn && (<Redirect to="/search" />)}`